translator/test/src/test/resources/features/TranslatorUnitTests.feature

```diff
@@ -152,4 +152,171 @@ Feature: Translator Service
+    When I try to translate jms message to kura data message
+    Then I got kura data message channel with "kapua-sys" scope, "rpione3" client id and proper semanticPart
+      | DEPLOY-V2    |
+      | GET          |
+      | packages     |
+    And I got kura data message with "byte[]" payload body
+    And No exception was thrown
+
+  Scenario: Translating of jms message with valid payload and valid topic into kura data message
+  Creating jms message with valid payload and valid topic. Trying to translate it into kura data message.
+  Check if jms message with valid topic and encoded payload body is received.
+
+    Given I create jms message with valid payload "response.code" and valid topic "kapua-sys.rpione3.DEPLOY-V2.GET.packages"
+    When I try to translate jms message to kura data message
+    Then I got kura data message channel with "kapua-sys" scope, "rpione3" client id and proper semanticPart
+      | DEPLOY-V2    |
+      | GET          |
+      | packages     |
+    And I got kura data message with proper payload metrics response code 200
+    And No exception was thrown
+
+  Scenario: Translating of jms message with empty payload and valid topic into kura data message
+  Creating jms message with empty payload and valid topic. Trying to translate it into kura data message.
+  Check if jms message with valid topic and empty payload body is received.
+
+    Given I create jms message with empty payload "" and valid topic "kapua-sys.rpione3.DEPLOY-V2.GET.packages"
+    And I try to translate jms message to kura data message
+    And I got kura data message channel with "kapua-sys" scope, "rpione3" client id and proper semanticPart
+      | DEPLOY-V2    |
+      | GET          |
+      | packages     |
+    Then I got kura data message with empty payload
+    And No exception was thrown
+
+  Scenario: Translating of jms message with empty payload and invalid topic that contain only userName into kura data message
+  Creating jms message with empty payload and invalid topic. Trying to translate it into kura data message.
+  Invalid channel exception should be thrown.
+
+    Given I create jms message with empty payload "" and valid topic "kapua-sys"
+    Then I expect the exception "InvalidChannelException" with the text " Invalid channel: kapua-sys"
+    When I try to translate jms message to kura data message
+    And An exception was thrown
+
+  Scenario: Translating invalid jms data message with valid channel, body and metrics into kura data message
+  Creating null jms message with valid channel, body and metrics. Trying to translate it into invalid jms message.
+  Check if mqtt message with valid topic and encoded body is received.
+  Invalid message exception should be thrown
+
+    Given I create jms message with valid payload "response.code" and valid topic "kapua-sys.rpione3.DEPLOY-V2.GET.packages"
+    Then I expect the exception "InvalidMessageException" with the text "Invalid message: null"
```

Comment:
Maybe this is something I need  to check. 
There should be a meaningful message